src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,175 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Assertation 1: `type` is a necessary property and has four possible values: `apiKey`, `HTTP`, `oauth2`, `openIdConnect`
+// Assertation 2: name property is required for `apiKey` type
+// Assertation 3: `in` property is required for `apiKey` type, valid values are: `query`, `header` or `cookie`
+// Assertation 4: `scheme` property` is required for `http` type
+// Assertation 5: `flows` object is required for `oauth2` type
+// Assertation 6: `opedIdConnectUrl` property is required for `openIdConnect` and must be a valid url
+
+const validator = require('validator');
+
+module.exports.validate = function({ resolvedSpec }) {
+  const API_KEY = 'apiKey';
+  const OAUTH2 = 'oauth2';
+  const HTTP = 'http';
+  const OPENID_CONNECT = 'openIdConnect';
+  const authTypes = [API_KEY, HTTP, OAUTH2, OPENID_CONNECT];
+  const PASSWORD = 'password';
+  const CLIENT_CREDENTIALS = 'client_Credentials';
+  const AUTHORIZATION_CODE = 'authorizationCode';
+  const errors = [];
+  const warnings = [];
+  const securitySchemes =
+    resolvedSpec.components && resolvedSpec.components.securitySchemes;
+
+  for (const key in securitySchemes) {
+    const path = `securitySchemes.${key}`;
+    const security = securitySchemes[key];
+    const type = security.type;
+
+    if (!type) {
+      errors.push({
+        message: 'security scheme is missing required field `type`',
+        path
+      });
+    } else if (authTypes.indexOf(type) === -1) {
+      errors.push({
+        message:
+          '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`',
+        path
+      });
+    } else {
+      //apiKey validation
+      if (type === API_KEY) {
+        const authIn = security.in;
+        if (!authIn || !['query', 'header', 'cookie'].includes(authIn)) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path
+          });
+        }
+        if (!security.name) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'name' string property. The name of the header or query parameter to be used.",
+            path
+          });
+        }
+      }
+      // oauth2 validation
+      else if (type === OAUTH2) {
+        const flows = security.flows;
+        const tokenUrl = security.tokenUrl;
```

Comment:
I don't think the [Security Scheme Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#security-scheme-object) has a `tokenUrl` property.

Reply:
Correct. The [OAuth Flow Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oauthFlowObject) does